diff --git Eigen/src/Core/products/GeneralBlockPanelKernel.h Eigen/src/Core/products/GeneralBlockPanelKernel.h
--- tmp/Eigen/src/Core/products/GeneralBlockPanelKernel.h
+++ tmp/Eigen/src/Core/products/GeneralBlockPanelKernel.h
@@ -1392,7 +1392,7 @@
           // NOTE: the begin/end asm comments below work around bug 935!
           // but they are not enough for gcc>=6 without FMA (bug 1637)
           #if EIGEN_GNUC_AT_LEAST(6,0)
-            #define EIGEN_GEBP_2PX4_SPILLING_WORKAROUND __asm__  ("" : [a0] "+rm" (A0),[a1] "+rm" (A1));
+            #define EIGEN_GEBP_2PX4_SPILLING_WORKAROUND __asm__  ("" : [a0] "+x,m" (A0),[a1] "+x,m" (A1));
           #else
             #define EIGEN_GEBP_2PX4_SPILLING_WORKAROUND
           #endif
